shrinkwrap.py

```diff
@@ -460,7 +460,8 @@ def download(args, root):
             else:
                 # This isn't a snap, pull the resource from the appropriate store
                 # use the bundle provided resource revision if available
-                resource_rev = app["resources"].get(resource.name) or resource.revision
+                res = app.get("resources", None)
+                resource_rev = res.get(resource.name) if res else resource.revision
```

Comment:
I suppose that there was an app without a `resources` key?

how about this instead:

```suggestion
                resource_rev = app.get("resources", {}).get(resource.name)
                resource_rev = resource_rev or resource.revision
```

there's a chance that if either key doesn't exist in this `app["resources"][resource.name]` we want to fall back to `resource.revision`